build/ninja_gen/src/git.rs

```diff
@@ -10,21 +10,31 @@ use crate::input::BuildInput;
 
 pub struct SyncSubmodule {
     pub path: &'static str,
+    pub offline_build: bool,
 }
 
 impl BuildAction for SyncSubmodule {
     fn command(&self) -> &str {
-        "git -c protocol.file.allow=always submodule update --init $path"
+        if self.offline_build {
+            "echo OFFLINE_BUILD is set, skipping git repository update for $path"
+        } else {
+            "git -c protocol.file.allow=always submodule update --init $path"
+        }
     }
 
     fn files(&mut self, build: &mut impl build::FilesHandle) {
-        if let Some(head) = locate_git_head() {
-            build.add_inputs("", head);
+        if self.offline_build {
+            build.add_variable("path", self.path);
+            build.add_output_stamp(format!("git/{}", self.path));
```

Comment:
These lines are repeated in the else branch, and can be placed outside instead. The else branch can then be removed, and this condition inverted.

Reply:
Indeed, thanks for the heads-up!